pkg/messageformat/compile_test.go

```diff
@@ -0,0 +1,329 @@
+package messageformat
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Compile(t *testing.T) {
+	t.Parallel()
+
+	for _, test := range []struct {
+		expectedErr error
+
+		name     string
+		expected string
+		input    AST
+	}{
+		{
+			name:        "error, empty node expression",
+			input:       AST{NodeExpr{}},
+			expectedErr: errors.New("expression node must not be empty"),
+		},
+		{
+			name: "error, no selectors",
+			input: AST{
+				NodeMatch{
+					Variants: []NodeVariant{
+						{
+							Keys: []string{"1"},
+						},
+					},
+				},
+			},
+			expectedErr: errors.New("there must be at least one selector"),
+		},
+		{
+			name: "error, mismatching number of keys and selectors",
+			input: AST{
+				NodeMatch{
+					Selectors: []NodeExpr{{Value: NodeVariable{Name: "one"}, Function: NodeFunction{Name: "func"}}},
+					Variants: []NodeVariant{
+						{
+							Keys: []string{"1", "2"},
+						},
+					},
+				},
+			},
+			expectedErr: errors.New("number of keys '2' for variant #0 don't match number of match selectors '1'"),
+		},
+		{
+			name:        "no nodes",
+			input:       AST{},
+			expected:    "",
```

Comment:
```suggestion
```

Reply:
I think it's best to leave it there for clarity.